app/apptesting/test_suite.go

```diff
@@ -99,3 +106,100 @@ func (keeperTestHelper *KeeperTestHelper) EndBlock() {
 	reqEndBlock := abci.RequestEndBlock{Height: keeperTestHelper.Ctx.BlockHeight()}
 	keeperTestHelper.App.EndBlocker(keeperTestHelper.Ctx, reqEndBlock)
 }
+
+func (keeperTestHelper *KeeperTestHelper) AllocateRewardsToValidator(valAddr sdk.ValAddress) {
+	validator, found := keeperTestHelper.App.StakingKeeper.GetValidator(keeperTestHelper.Ctx, valAddr)
+	keeperTestHelper.Require().True(found)
+
+	// allocate reward tokens to distribution module
+	coins := sdk.Coins{sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(20000))}
+	err := simapp.FundModuleAccount(keeperTestHelper.App.BankKeeper, keeperTestHelper.Ctx, distrtypes.ModuleName, coins)
+	keeperTestHelper.Require().NoError(err)
+
+	// allocate rewards to validator
+	keeperTestHelper.Ctx = keeperTestHelper.Ctx.WithBlockHeight(keeperTestHelper.Ctx.BlockHeight() + 1)
+	decTokens := sdk.DecCoins{{Denom: sdk.DefaultBondDenom, Amount: sdk.NewDec(20000)}}
+	keeperTestHelper.App.DistrKeeper.AllocateTokensToValidator(keeperTestHelper.Ctx, validator, decTokens)
+	keeperTestHelper.App.DistrKeeper.IncrementValidatorPeriod(keeperTestHelper.Ctx, validator)
```

Comment:
why is this changing the validator period? I don't think it should be, the validators delegation amount isn't changing

Reply:
Oh is it ran upon delegation amount change? I thought it was running based on reward change on validators

Reply:
nope, delegation amount change only!